web/src/sections/actions/ToolsList.tsx

```diff
@@ -3,71 +3,124 @@
-  tools,
+  isFetching = false,
+  onRetry,
+  totalCount,
+  enabledCount,
+  showOnlyEnabled = false,
+  onToggleShowOnlyEnabled,
+  isEmpty = false,
   searchQuery,
-  onToolToggle,
+  emptyMessage = "No tools available",
+  emptySearchMessage = "No tools found",
+  children,
   className,
-  isFetching,
-  onRetry,
 }) => {
   const hasRetried = useRef(false);
 
   useEffect(() => {
     // If the server reports tools but none were returned, try one automatic refetch
-    if (!isFetching && tools.length === 0 && !hasRetried.current && onRetry) {
+    if (!isFetching && isEmpty && !hasRetried.current && onRetry) {
       hasRetried.current = true;
       onRetry();
     }
-  }, [isFetching, tools.length, onRetry]);
+  }, [isFetching, isEmpty, onRetry]);
+
+  const showFooter =
+    totalCount !== undefined && enabledCount !== undefined && totalCount > 0;
 
   return (
-    <div
-      className={cn(
-        "flex flex-col gap-1 items-start max-h-[480px] overflow-y-auto w-full",
-        className
-      )}
-    >
-      {isFetching ? (
-        // Show 5 skeleton items while loading
-        <>
-          {[...Array(5)].map((_, index) => (
-            <ToolItemSkeleton key={`skeleton-${index}`} />
-          ))}
-        </>
-      ) : tools.length > 0 ? (
-        tools.map((tool) => (
-          <ToolItem
-            key={tool.id}
-            name={tool.name}
-            description={tool.description}
-            icon={tool.icon}
-            isAvailable={tool.isAvailable}
-            isEnabled={tool.isEnabled}
-            onToggle={(enabled) => onToolToggle?.(tool.id, enabled)}
-          />
-        ))
-      ) : (
-        <div className="flex items-center justify-center w-full py-8">
-          <Text text03 mainUiBody>
-            {searchQuery ? "No tools found" : "No tools available"}
-          </Text>
+    <>
+      <div
+        className={cn(
+          "flex flex-col gap-1 items-start max-h-[480px] overflow-y-auto w-full",
+          className
+        )}
+      >
+        {isFetching ? (
+          // Show 5 skeleton items while loading
+          <>
+            {[...Array(5)].map((_, index) => (
+              <ToolItemSkeleton key={`skeleton-${index}`} />
+            ))}
+          </>
+        ) : isEmpty ? (
+          // Empty state
+          <div className="flex items-center justify-center w-full py-8">
+            <Text text03 mainUiBody>
+              {searchQuery ? emptySearchMessage : emptyMessage}
+            </Text>
+          </div>
+        ) : (
+          // Render children (actual tool items)
+          children
+        )}
+      </div>
+
+      {/* Footer showing enabled tool count with filter toggle */}
+      {showFooter && !isEmpty && !isFetching && (
+        <div className="relative w-full">
+          {/* Gradient fade overlay */}
+          <div className="absolute inset-x-0 -top-8 h-8 bg-gradient-to-b from-transparent to-background pointer-events-none" />
```

Comment:
Could we convert this into its own component? I think there are locations where I could use this as well!

Reply:
Usage could look like:

```tsx
{showFooter && !isEmpty && !isFetching && (
  <FadeDiv>
    {children}
  </FadeDiv>
)}
```